homeassistant/components/nasweb/__init__.py

```diff
@@ -0,0 +1,151 @@
+    NASWEB_CONFIG_URL,
+)
+from .coordinator import NASwebCoordinator
+from .nasweb_data import NASwebData
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up NASweb from a config entry."""
+
+    if DOMAIN not in hass.data:
+        data = NASwebData()
+        data.initialize(hass)
+        hass.data[DOMAIN] = data
+    nasweb_data: NASwebData = hass.data[DOMAIN]
+
+    webio_api = WebioAPI(
+        entry.data[CONF_HOST], entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD]
+    )
+    try:
+        if not await webio_api.check_connection():
+            raise ConfigEntryNotReady(
+                f"[{entry.data[CONF_HOST]}] Check connection failed"
+            )
+        if not await webio_api.refresh_device_info():
+            _LOGGER.error("[%s] Refresh device info failed", entry.data[CONF_HOST])
+            raise MissingNASwebData
+        webio_serial = webio_api.get_serial_number()
+        if webio_serial is None:
+            _LOGGER.error("[%s] Serial number not available", entry.data[CONF_HOST])
+            raise MissingNASwebData
+
+        coordinator = NASwebCoordinator(
+            hass, webio_api, name=f"NASweb[{webio_api.get_name()}]"
+        )
+        nasweb_data.entries_coordinators[entry.entry_id] = coordinator
+        nasweb_data.notify_coordinator.add_coordinator(webio_serial, coordinator)
+
+        webhook_url = nasweb_data.get_webhook_url(hass)
+        if webhook_url is None:
+            _LOGGER.error("Cannot pass Home Assistant url to NASweb device")
+            raise MissingNASwebData
+        if not await webio_api.status_subscription(webhook_url, True):
+            _LOGGER.error("Failed to subscribe for status updates from webio")
+            raise MissingNASwebData
+    except AuthError:
+        ir.async_create_issue(
+            hass,
+            DOMAIN,
+            ISSUE_INVALID_AUTHENTICATION,
+            is_fixable=False,
+            is_persistent=False,
+            severity=ir.IssueSeverity.ERROR,
+            translation_key=ISSUE_INVALID_AUTHENTICATION,
+            translation_placeholders={
+                "domain_name": DOMAIN_DISPLAY_NAME,
+                "device_name": entry.title,
+            },
+        )
+        return False
+    except MissingNASwebData:
+        ir.async_create_issue(
+            hass,
+            DOMAIN,
+            ISSUE_INTERNAL_ERROR,
+            is_fixable=False,
+            is_persistent=False,
+            severity=ir.IssueSeverity.ERROR,
+            translation_key=ISSUE_INTERNAL_ERROR,
+            translation_placeholders={
+                "domain_name": DOMAIN_DISPLAY_NAME,
+                "device_name": entry.title,
+            },
+        )
+        return False
+    ir.async_delete_issue(hass, DOMAIN, ISSUE_INVALID_AUTHENTICATION)
+    ir.async_delete_issue(hass, DOMAIN, ISSUE_INTERNAL_ERROR)
```

Comment:
I decided to remove the issues because they are still present if integration is reloaded successfully after issue is created. I'm not supposed to remove them like that? Should I leave them for now and get back to removing them when repair flow is added in future PR's?

Reply:
> I decided to remove the issues because they are still present if integration is reloaded successfully after issue is created

When can that happen without the repair flow?